File: ltrim/ltrim.java
import java.util.regex.Pattern;

public class ltrim {
    private final static Pattern LTRIM = Pattern.compile("^\\s+");

    public static String ltrim(String s) {
        return LTRIM.matcher(s).replaceAll("");
    }
    public static void main(String args[]) {

        // ATOMIC_FUNCTION:  trim($val)
        // TAGS: string
        // LANG:  java
        // FUNCTION: trim($val)

        String start_val = "  abcdefg  ";
        String expected_end_val = "abcdefg  ";
        String end_val = ltrim(start_val);



        // TEST
        if(end_val.equals(expected_end_val)){
            System.out.println("PASS");
        } else {
            System.out.println("FAIL");
        }
    }
}

File: ltrim/ltrim.js
// ATOMIC_FUNCTION:  ltrim($val)
// TAGS: string
// LANG:  node.js
// FUNCTION: ltrim($val) 
function ltrim(str, chr) {
    var rgxtrim = (!chr) ? new RegExp('^\\s+') : new RegExp('^'+chr+'+');
    return str.replace(rgxtrim, '');
}

var start_val = "  abcdefg  ";
var expected_end_val = "abcdefg  ";
var end_val = ltrim(start_val);

// TEST
if(end_val == expected_end_val) {
	console.log("PASS");
} else {
	console.log("FAIL");
}
